solutions/platinum/cf-1254D.mdx

```diff
@@ -0,0 +1,151 @@
+---
+id: cf-1254D
+source: CF
+title: Tree Queries
+author: Rameez Parwez
+---
+
+[Official Editorial (C++)](https://codeforces.com/blog/entry/71594)
+
+## Explanation
+
+The idea is to use Heavy-Light Decomposition(HLD) for its efficiency in dividing the tree into chains and ensuring that paths between any two nodes pass through a limited number of chains, along with Binary Indexed Tree(BIT) to efficiently manage cumulative sums and range updates operations.
+
+At first, the DFS is performed starting from node $0$ to calculate the depth and subtree size. Afterwards, the decomposition is performed to transform the tree into a structure suitable for BIT operations.
+
+For a modification, we first check if vertex $v$ has some subtree. If it does, we want to add a tag to $v$ which increase the subtree of every light child $u$ of $v$ by $(n - x) * d$. When handling the query, we find its value on BIT and sum of all tags on its path to the root.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O} (N + Q \log(N))$
+
+<LanguageSection>
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+const int MAX_N = 150001, MOD = 998244353;
+vector<int> adj[MAX_N], sum(MAX_N);
+vector<int> top(MAX_N), in(MAX_N), parent(MAX_N), depth(MAX_N), out(MAX_N),
```

Comment:
whats the point of making these vectors if they are at a constant size anyways?

Reply:
It's just a preference, if you want me to use an array then I will convert it